### PR TITLE
(WIP) Add feature for polarity using SentiWordnet

### PR DESCRIPTION
**Pull request by codez266**
Adds a library sentiment_classifier that makes use of SentiWordnet and
WSD to detect the polarity of each word in a document and returns an
aggregate positive and negative score per document. Add these polarity
scores as features to enwiki.py

See - https://phabricator.wikimedia.org/T167305

### CODE REVIEW

File: draftquality/feature_lists/enwiki.py [hunk 1]
```diff
@@ -1,6 +1,40 @@
...
+from revscoring.datasources import Datasource
+from senti_classifier import senti_classifier
+
+def get_polarity_score(revision_text):
+    """
+    Gets the positive and negative polarity of the document using SentiWordnet
+    limits the document to 20 sentences for efficiency
+    """
+    doc = revision_text.lower().split("\n")
+    return senti_classifier.polarity_scores(doc[:20])
```
Comment (halfak) [inline draftquality/feature_lists/enwiki.py]: Only the first 20 chars?

**Review by halfak — CHANGES_REQUESTED** [review]

File: draftquality/feature_lists/enwiki.py (hunk 1 above)
Comment (codez266) [reply draftquality/feature_lists/enwiki.py]: First 20 sentences. The polarity_scores method by default takes a list of sentences as a document. Hence, I'm first splitting the text at '\n' and passing the first 20 sentences as input.
See - https://github.com/kevincobain2000/sentiment_classifier/#library-usage

Comment (codez266) [reply draftquality/feature_lists/enwiki.py]: I'll see if I can make any quick performance improvements and ping again, before you test on a larger dataset.

**Comment by codez266** [comment]
Now taking only the most common word sense for polarity to improve computation. Ready for testing on a bigger dataset.

**Comment by adamwight** [comment]
@codez266 fyi, I'm still struggling to learn the toolchain.  I've run small 10k samples through training and testing, but pretty sure I did something wrong or simply don't understand what's happening, because it claims an accuracy of 94%.  See https://phabricator.wikimedia.org/P5648

File: draftquality/feature_lists/enwiki.py [hunk 2]
```diff
@@ -1,6 +1,48 @@
...
+from revscoring import Feature
+from revscoring.datasources import Datasource
+from nltk.corpus import sentiwordnet as swn
+
+def get_polarity_score(non_stop_tokens):
+    """
+    Gets the positive and negative polarity of the document using SentiWordnet
+    takes the most common sense of the word for efficiency
+    """
+    pos, neg = 0, 0
```
Comment (adamwight) [inline draftquality/feature_lists/enwiki.py:13]: I had to change these to 0.0 in order to get floats out.

File: draftquality/feature_lists/enwiki.py [hunk 3]
```diff
@@ -143,5 +185,12 @@
     max(wikitext.revision.content_chars, 1)
 ]
 
+sentiment_based = [
+    positive_polarity,
+    negative_polarity,
+    diff_polarity
+]
+
 draft_quality = (char_based + token_based + parse_based +
```
Comment (adamwight) [inline draftquality/feature_lists/enwiki.py:194]: An extremely tiny thing I noticed for next time: It's helpful to leave the baseline feature list unchanged, and add a second feature list like "draft_quality_with_sentiment" for ease of building a side-by-side comparison.

File: draftquality/feature_lists/enwiki.py [hunk 4]
```diff
@@ -1,6 +1,48 @@
 from revscoring.features import wikitext
 from revscoring.features.modifiers import max, sub
 from revscoring.languages import english
+from revscoring import Feature
+from revscoring.datasources import Datasource
+from nltk.corpus import sentiwordnet as swn
```
Comment (adamwight) [inline draftquality/feature_lists/enwiki.py:6]: Please add a note in the README, letting the user know to download the sentiwordnet nltk package...

**Review by adamwight — COMMENTED** [review]
Looking really good!  I'm still running the model health check...

**Comment by Pix1234** [comment]
author's request

**Pull request closed without merging**